pilot/pkg/xds/workloadentry.go

```diff
@@ -0,0 +1,219 @@
+// Copyright Istio Authors
```

Comment:
Before merge would be great to have unit tests for this. Lots of edge cases that should be reasonable to capture in  unit tests